src/LintDirtyModulesPlugin.js

```diff
@@ -22,12 +21,8 @@ export default class LintDirtyModulesPlugin {
       return;
     }
 
-    const { cli } = getCLIEngine(this.options);
     const dirtyOptions = { ...this.options };
-    const glob = cli
-      .resolveFileGlobPatterns(dirtyOptions.files)
-      .join('|')
-      .replace(/\\/g, '/');
+    const glob = dirtyOptions.files.join('|').replace(/\\/g, '/');
```

Comment:
We will have problems here. We need to find a solution to `resolveFileGlobPatterns`
https://github.com/webpack-contrib/eslint-webpack-plugin/issues/2
https://github.com/webpack-contrib/eslint-webpack-plugin/commit/d5c8820d9467e8794a4aa3944bf6ded746d79411

Reply:
Thank you for the links, I will investigate further.

Reply:
Quick update as I have been silent all week. Thank you for pointing out the issue.

It is indeed causing an issue when watching folder paths with the `lintDirtyModulesOnly` option enabled.

I have added tests and reimplemented the functionality of `resolveFileGlobPatterns`. I am still refactoring and rebasing, it should be ready for review in a day or two.